allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -269,6 +339,57 @@ def forward(self,  # type: ignore
             # TODO(matt): compute accuracy here.
             return outputs
 
+    def _compute_average_with_encoder(self, embed_tensor: Variable,
+                                      mask: Variable) -> Variable:
+        # computes the average along dimension 1 of the embed_tensor
+        dims = embed_tensor.size()
+        table_mask = mask.view(-1, mask.size(2))
+        averaged_tensor = self._entity_encoder(embed_tensor.view(-1, dims[2], dims[3]),
+                                               table_mask)
+        # (batch_size, num_entities, embedding_dim)
+        return averaged_tensor.view(dims[0], dims[1], dims[3])
+
+    def _get_neighbor_indexes(self, world: List[WikiTablesWorld],
+                                    max_num_entities: int) -> Variable:
+        # returns each entity's neighbor indexes
+        batches = []
```

Comment:
This is processing the instances in a single batch, not multiple batches.  And `batches` as a variable name doesn't really say anything about what the variable contains, only that it's a collection.  Something like `batch_neighbors` would be more informative here.